src/main/java/com/impactupgrade/nucleus/util/PageResultsProcessor.java

```diff
@@ -0,0 +1,37 @@
+package com.impactupgrade.nucleus.util;
+
+import com.google.common.base.Strings;
+import com.impactupgrade.nucleus.model.PagedResults;
+
+import java.util.function.Consumer;
+
+public class PageResultsProcessor<T> {
+
+  private final Consumer<PagedResults.ResultSet<T>> consumer;
+  private final CheckedFunction<String, PagedResults.ResultSet<T>> getter;
+
+  public PageResultsProcessor(Consumer<PagedResults.ResultSet<T>> consumer,
+                              CheckedFunction<String, PagedResults.ResultSet<T>> getter) {
+    this.consumer = consumer;
+    this.getter = getter;
+  }
+
+  public void processPagedResults(PagedResults<T> pagedResults) throws Exception {
+    for (PagedResults.ResultSet<T> resultSet : pagedResults.getResultSets()) {
+      processResultSet(resultSet);
+    }
+  }
+
+  private void processResultSet(PagedResults.ResultSet<T> resultSet) throws Exception {
+    consumer.accept(resultSet);
```

Comment:
@brmeyer Is it possible that we may need to have 2 types of processing: "by page" vs. "by result set"?

I mean we can call the consumer function inside for-each page vs calling it recursively for-each result set (for example based on processor setting, like enum or similar to that)